newrelic/resource_newrelic_nrql_alert_condition.go

```diff
@@ -85,74 +81,221 @@ func resourceNewRelicNrqlAlertCondition() *schema.Resource {
+				MinItems:    1,
+				MaxItems:    2,
+				Description: "A set of terms for this condition. Max 2 terms allowed - at least one 1 critical term and 1 optional warning term.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						// Maps to `thresholdDuration` in NerdGraph and values are in seconds, not minutes.
+						// Validation is different in NerdGraph - Value must be within 120-3600 seconds (2-60 minutes) and a multiple of 60 for BASELINE conditions.
+						// Convert to seconds when using NerdGraph
 						"duration": {
+							Deprecated:   "use `threshold_duration` attribute instead",
 							Type:         schema.TypeInt,
-							Required:     true,
-							ValidateFunc: validation.IntBetween(1, 120),
+							Optional:     true,
 							Description:  "In minutes, must be in the range of 1 to 120, inclusive.",
+							ValidateFunc: validation.IntBetween(1, 120),
 						},
+						// Value must be uppercase when using NerdGraph
 						"operator": {
 							Type:         schema.TypeString,
 							Optional:     true,
 							Default:      "equal",
-							ValidateFunc: validation.StringInSlice([]string{"above", "below", "equal"}, false),
 							Description:  "One of (above, below, equal). Defaults to equal.",
+							ValidateFunc: validation.StringInSlice([]string{"above", "below", "equal"}, false),
 						},
+						// Value must be uppercase when using NerdGraph
 						"priority": {
 							Type:         schema.TypeString,
 							Optional:     true,
 							Default:      "critical",
+							Description:  "One of (critical, warning). Defaults to critical. At least one condition term must have priority set to critical.",
 							ValidateFunc: validation.StringInSlice([]string{"critical", "warning"}, false),
-							Description:  "One of (critical, warning). Defaults to critical.",
 						},
 						"threshold": {
 							Type:         schema.TypeFloat,
 							Required:     true,
+							Description:  "Must be 0 or greater. For baseline conditions must be in range [1, 1000].",
 							ValidateFunc: float64Gte(0.0),
-							Description:  "Must be 0 or greater.",
 						},
+						// Does not exist in NerdGraph
 						"time_function": {
+							Deprecated:   "use `threshold_occurrences` attribute instead",
 							Type:         schema.TypeString,
-							Required:     true,
+							Optional:     true,
+							Description:  "Valid values are: 'all' or 'any'",
 							ValidateFunc: validation.StringInSlice([]string{"all", "any"}, false),
-							Description:  "One of (all, any).",
+						},
+
+						// NerdGraph only. Seems to be similar to `time_function`
+						"threshold_occurrences": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Valid values are: 'ALL' or 'AT_LEAST_ONCE'",
+							ValidateFunc: validation.StringInSlice([]string{"ALL", "AT_LEAST_ONCE"}, false),
+						},
+						// NerdGraph only. Equivalent to `duration`, but in seconds
+						"threshold_duration": {
```

Comment:
See [comment](https://github.com/terraform-providers/terraform-provider-newrelic/pull/487/files#r403361992) above 😎

Reply:
Thanks to some help with from the HashiCorp crew, it turns out this can be handled the same way it's handled in the `nrql` block even though that's a `TypeList`.